src/components/swap/SwapModalFooter.tsx

```diff
@@ -1,36 +1,130 @@
 import { Trans } from '@lingui/macro'
-import { Trade } from '@uniswap/router-sdk'
-import { Currency, TradeType } from '@uniswap/sdk-core'
+import { Currency, CurrencyAmount, Percent, Token, TradeType } from '@uniswap/sdk-core'
+import { ElementName, EventName } from 'components/AmplitudeAnalytics/constants'
+import { Event } from 'components/AmplitudeAnalytics/constants'
+import { TraceEvent } from 'components/AmplitudeAnalytics/TraceEvent'
+import { useStablecoinValue } from 'hooks/useStablecoinPrice'
+import useTransactionDeadline from 'hooks/useTransactionDeadline'
 import { ReactNode } from 'react'
 import { Text } from 'rebass'
+import { InterfaceTrade } from 'state/routing/types'
+import { useClientSideRouter, useUserSlippageTolerance } from 'state/user/hooks'
+import { computeRealizedLPFeePercent } from 'utils/prices'
 
 import { ButtonError } from '../Button'
 import { AutoRow } from '../Row'
+import { getPriceImpact } from './AdvancedSwapDetails'
 import { SwapCallbackError } from './styleds'
 
+function getDurationTillTimestampSinceEpoch(futureTimestampSinceEpoch?: number): number | undefined {
+  if (!futureTimestampSinceEpoch) return undefined
+  return futureTimestampSinceEpoch - new Date().getTime() / 1000
+}
+
+const getNumberFormattedToDecimalPlace = (
+  intialNumberObject: Percent | CurrencyAmount<Token | Currency>,
+  decimalPlace: number
+): number => parseFloat(intialNumberObject.toFixed(decimalPlace))
+
+interface AnalyticsEventProps {
+  trade: InterfaceTrade<Currency, Currency, TradeType>
+  txHash: string | undefined
+  allowedSlippage: Percent
+  transactionDeadlineSecondsSinceEpoch: number | undefined
+  isAutoSlippage: boolean
+  isAutoRouterApi: boolean
+  tokenInAmountUsd: string | undefined
+  tokenOutAmountUsd: string | undefined
+  lpFeePercent: Percent
+}
+
+const formatAnalyticsEventProperties = ({
+  trade,
+  txHash,
+  allowedSlippage,
+  transactionDeadlineSecondsSinceEpoch,
+  isAutoSlippage,
+  isAutoRouterApi,
+  tokenInAmountUsd,
+  tokenOutAmountUsd,
+  lpFeePercent,
+}: AnalyticsEventProps) => ({
+  estimated_network_fee_usd: trade.gasUseEstimateUSD
+    ? getNumberFormattedToDecimalPlace(trade.gasUseEstimateUSD, 2)
+    : undefined,
+  transaction_hash: txHash,
+  transaction_deadline_seconds: getDurationTillTimestampSinceEpoch(transactionDeadlineSecondsSinceEpoch),
+  token_in_amount_usd: tokenInAmountUsd ? parseFloat(tokenInAmountUsd) : undefined,
+  token_out_amount_usd: tokenOutAmountUsd ? parseFloat(tokenOutAmountUsd) : undefined,
+  token_in_address: trade.inputAmount.currency.isToken ? trade.inputAmount.currency.address : undefined,
+  token_out_address: trade.outputAmount.currency.isToken ? trade.outputAmount.currency.address : undefined,
+  token_in_symbol: trade.inputAmount.currency.symbol,
+  token_out_symbol: trade.outputAmount.currency.symbol,
+  token_in_amount: getNumberFormattedToDecimalPlace(trade.inputAmount, trade.inputAmount.currency.decimals),
```

Comment:
can you use the same function as we have in `Swap/index.tsx`? look for `formattedAmounts`

Reply:
that isn't an exportable function, it uses hooks and has a lot of dependencies in the Swap/index.tsx page. I don't think it is appropriate to replicate all of that logic here (seems like overkill)